### ISSUE TITLE
~/Documents directory crashes nemo immediately

### ISSUE THREAD

**Issue by michaelwhitford** [linuxmint/nemo#2452]
```
 * Nemo version (nemo --version)
 * Is issue with desktop or windowed nemo?
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
 * Graphics hardware *and* driver used
 * 32 or 64 bit
 ```
* nemo 4.6.4
* Windowed mode
* Linux Mint 20 (fresh install)
* Nvidia RTX 2080 - driver v440
* 64 bit

**Issue**

Opening Documents directory from home crashes nemo as soon as I open it.  I can open any other directory with no issues, but as soon as I try to enter documents it crashes (stack trace below).  I have used the utility detox to remove any special characters from files in ~/Documents, I feared the problem was a non-printable character.  I even removed spaces.

**Steps to reproduce**

Open Documents directory

**Expected behaviour**

Display directory contents

**Other information**
$ RUST_BACKTRACE=full nemo ~/Documents
Nemo-Share-Message: 15:10:01.442: Called "net usershare info" but it failed: 'net usershare' returned error 255: net usershare: cannot open usershare directory /var/lib/samba/usershares. Error No such file or directory
Please ask your system administrator to enable user sharing.

thread '<unnamed>' panicked at 'Cairo error "cairo_restore() without matching cairo_save()"', /build/librsvg-zD3s3K/librsvg-2.48.7/vendor/cairo-rs/src/enums.rs:274:13
stack backtrace:
   0:     0x7f3c9346db93 - <unknown>
   1:     0x7f3c934ac6fd - <unknown>
   2:     0x7f3c9346a2e5 - <unknown>
   3:     0x7f3c9347560e - <unknown>
   4:     0x7f3c93475320 - <unknown>
   5:     0x7f3c93475dab - <unknown>
   6:     0x7f3c9347595e - <unknown>
   7:     0x7f3c934758a0 - <unknown>
   8:     0x7f3c931410a0 - <unknown>
   9:     0x7f3c930f8176 - <unknown>
  10:     0x7f3c93031a05 - <unknown>
  11:     0x7f3c93085173 - <unknown>
  12:     0x7f3c930317dc - <unknown>
  13:     0x7f3c9308743a - <unknown>
  14:     0x7f3c93031c6a - <unknown>
  15:     0x7f3c931198a5 - <unknown>
  16:     0x7f3c93082151 - <unknown>
  17:     0x7f3c9311984f - <unknown>
  18:     0x7f3c93031a05 - <unknown>
  19:     0x7f3c93085173 - <unknown>
  20:     0x7f3c930317dc - <unknown>
  21:     0x7f3c9308743a - <unknown>
  22:     0x7f3c93031c6a - <unknown>
  23:     0x7f3c931198a5 - <unknown>
  24:     0x7f3c93081fec - <unknown>
  25:     0x7f3c9311984f - <unknown>
  26:     0x7f3c93031a05 - <unknown>
  27:     0x7f3c93085173 - <unknown>
  28:     0x7f3c930317dc - <unknown>
  29:     0x7f3c9308743a - <unknown>
  30:     0x7f3c93031c6a - <unknown>
  31:     0x7f3c9311baea - <unknown>
  32:     0x7f3c93081fec - <unknown>
  33:     0x7f3c9311b92f - <unknown>
  34:     0x7f3c93031a05 - <unknown>
  35:     0x7f3c93085173 - <unknown>
  36:     0x7f3c930317dc - <unknown>
  37:     0x7f3c9308743a - <unknown>
  38:     0x7f3c930bb358 - <unknown>
  39:     0x7f3c9300e7d7 - <unknown>
  40:     0x7f3c930112c1 - rsvg_rust_handle_get_pixbuf_sub
  41:     0x7f3c993f5486 - <unknown>
  42:     0x7f3c993cfaa3 - gdk_pixbuf_loader_close
  43:     0x7f3c993cd00a - gdk_pixbuf_new_from_file_at_scale
  44:     0x55fb2226e95c - <unknown>
  45:     0x55fb2226e5a6 - <unknown>
  46:     0x7f3c98d3b0b3 - __libc_start_main
  47:     0x55fb2226e87e - <unknown>
  48:                0x0 - <unknown>
fatal runtime error: failed to initiate panic, error 5
thread '<unnamed>' panicked at 'Cairo error "cairo_restore() without matching cairo_save()"', /build/librsvg-zD3s3K/librsvg-2.48.7/vendor/cairo-rs/src/enums.rs:274:13
stack backtrace:
   0:     0x7f82b4841b93 - <unknown>
   1:     0x7f82b48806fd - <unknown>
   2:     0x7f82b483e2e5 - <unknown>
   3:     0x7f82b484960e - <unknown>
   4:     0x7f82b4849320 - <unknown>
   5:     0x7f82b4849dab - <unknown>
   6:     0x7f82b484995e - <unknown>
   7:     0x7f82b48498a0 - <unknown>
   8:     0x7f82b45150a0 - <unknown>
   9:     0x7f82b44cc176 - <unknown>
  10:     0x7f82b4405a05 - <unknown>
  11:     0x7f82b4459173 - <unknown>
  12:     0x7f82b44057dc - <unknown>
  13:     0x7f82b445b43a - <unknown>
  14:     0x7f82b4405c6a - <unknown>
  15:     0x7f82b44ed8a5 - <unknown>
  16:     0x7f82b4456151 - <unknown>
  17:     0x7f82b44ed84f - <unknown>
  18:     0x7f82b4405a05 - <unknown>
  19:     0x7f82b4459173 - <unknown>
  20:     0x7f82b44057dc - <unknown>
  21:     0x7f82b445b43a - <unknown>
  22:     0x7f82b4405c6a - <unknown>
  23:     0x7f82b44ed8a5 - <unknown>
  24:     0x7f82b4455fec - <unknown>
  25:     0x7f82b44ed84f - <unknown>
  26:     0x7f82b4405a05 - <unknown>
  27:     0x7f82b4459173 - <unknown>
  28:     0x7f82b44057dc - <unknown>
  29:     0x7f82b445b43a - <unknown>
  30:     0x7f82b4405c6a - <unknown>
  31:     0x7f82b44efaea - <unknown>
  32:     0x7f82b4455fec - <unknown>
  33:     0x7f82b44ef92f - <unknown>
  34:     0x7f82b4405a05 - <unknown>
  35:     0x7f82b4459173 - <unknown>
  36:     0x7f82b44057dc - <unknown>
  37:     0x7f82b445b43a - <unknown>
  38:     0x7f82b448f358 - <unknown>
  39:     0x7f82b43e27d7 - <unknown>
  40:     0x7f82b43e52c1 - rsvg_rust_handle_get_pixbuf_sub
  41:     0x7f82b4beb486 - <unknown>
  42:     0x7f82c9c82aa3 - gdk_pixbuf_loader_close
  43:     0x7f82ca724f26 - <unknown>
  44:     0x7f82ca725573 - gnome_desktop_thumbnail_factory_generate_thumbnail
  45:     0x556e8dca03e0 - <unknown>
  46:     0x7f82c9b46c62 - <unknown>
  47:     0x7f82c99821d4 - <unknown>
  48:     0x7f82c9981931 - <unknown>
  49:     0x7f82c942c609 - start_thread
  50:     0x7f82c9353103 - __clone
  51:                0x0 - <unknown>
fatal runtime error: failed to initiate panic, error 5
Aborted (core dumped)

**Comment by mtwebster** [comment]
Try deleting the thumbnail cache (`rm -rf ~/.cache/thumbnails`)

**Comment by mtwebster** [comment]
If you use xviewer and open that folder, are you able to view all image files with it?

**Comment by michaelwhitford** [comment]
I tried deleting the thumbnail cache, it did not work.  xviewer is crashing on a svg file I created in inkscape...  And moving that file to another directory fixed nemo.  This appears to be from a badly formed svg file.  It affected xviewer as well so maybe and issue with a common library instead of the app itself?

**Comment by mtwebster** [comment]
That's what it seems from the trace, though it's not really an excuse for crashing.  We run into these once in a while for different formats.  Would it be possible for you to share the svg file? That would allow me to try and reproduce and see exactly how it's crashing nemo (the thumbnailing takes place in a separate process, so if that crashes it would be ok, but the fact nemo crashes puzzles me).

**Comment by michaelwhitford** [comment]
I actually opened it and resaved it with the newest version of inkscape and fixed the file.  I did not save a copy of the broken svg.  I played a bit with saving various pieces of it in seperate files to try to reproduce but I could not get it to crash again.  I made the file about 7 years ago and just the latest mint 20 started crashing nemo from the file being in my documents folder.  On older versions of mint nemo did not crash from this file. Wish I could be more help.
